### PR TITLE
TINY-9669: Disable core toolbars/menuitems based on selection editable state

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket: TINY-9669

Description of Changes:
* Adds disabling toggling behavior when selecting noneditable contents for a ton of UI controls.
* There will be a follow up ticket to fix CSS issues some controls doesn't render correctly when being disabled.

Sorry for the fat PR but should be possible to review commit by commit.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/link/test/ts/browser/NoneditableRootTest.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Remove link"][aria-disabled="true"]');
+      TinySelections.setSelection(editor, [ 1, 0, 0 ], 0, [ 1, 0, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Remove link"][aria-disabled="false"]');
+    });
+  });
+
+  it('TINY-9669: Disable link menuitem on noneditable content', async () => {
+    const editor = hook.editor();
+    editor.getBody().contentEditable = 'false';
```
Comment (lorenzo-pomili) [inline modules/tinymce/src/plugins/link/test/ts/browser/NoneditableRootTest.ts:47]: use `withNoneditableRootEditor` if possible

File: modules/tinymce/src/plugins/link/test/ts/browser/NoneditableRootTest.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    TinySelections.setSelection(editor, [ 1, 0 ], 0, [ 1, 0 ], 2);
+    TinyUiActions.clickOnMenu(editor, 'button:contains("Insert")');
+    await TinyUiActions.pWaitForUi(editor, '[role="menuitem"][title="Link..."][aria-disabled="false"]');
+    TinyUiActions.keystroke(editor, Keys.escape());
+    editor.getBody().contentEditable = 'true';
+  });
+
+  it('TINY-9669: Disable unlink menuitem on noneditable content', async () => {
+    const editor = hook.editor();
+    editor.getBody().contentEditable = 'false';
```
Comment (lorenzo-pomili) [inline modules/tinymce/src/plugins/link/test/ts/browser/NoneditableRootTest.ts:62]: use `withNoneditableRootEditor` if possible

File: modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NoneditableRootTest.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Nonbreaking space"][aria-disabled="true"]');
+      TinySelections.setSelection(editor, [ 1, 0 ], 0, [ 1, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Nonbreaking space"][aria-disabled="false"]');
+    });
+  });
+
+  it('TINY-9669: Disable nonbreaking menuitem on noneditable content', async () => {
+    const editor = hook.editor();
+    editor.getBody().contentEditable = 'false';
```
Comment (lorenzo-pomili) [inline modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NoneditableRootTest.ts:34]: use `withNoneditableRootEditor` if possible

File: modules/tinymce/src/plugins/pagebreak/test/ts/browser/NoneditableRootTest.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Page break"][aria-disabled="true"]');
+      TinySelections.setSelection(editor, [ 1, 0 ], 0, [ 1, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Page break"][aria-disabled="false"]');
+    });
+  });
+
+  it('TINY-9669: Disable pagebreak menuitem on noneditable content', async () => {
+    const editor = hook.editor();
+    editor.getBody().contentEditable = 'false';
```
Comment (lorenzo-pomili) [inline modules/tinymce/src/plugins/pagebreak/test/ts/browser/NoneditableRootTest.ts:34]: use `withNoneditableRootEditor` if possible

File: modules/tinymce/src/plugins/template/test/ts/browser/NoneditableRootTest.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Insert template"][aria-disabled="true"]');
+      TinySelections.setSelection(editor, [ 1, 0 ], 0, [ 1, 0 ], 2);
+      UiFinder.exists(SugarBody.body(), '[aria-label="Insert template"][aria-disabled="false"]');
+    });
+  });
+
+  it('TINY-9669: Disable template menuitem on noneditable content', async () => {
+    const editor = hook.editor();
+    editor.getBody().contentEditable = 'false';
```
Comment (lorenzo-pomili) [inline modules/tinymce/src/plugins/template/test/ts/browser/NoneditableRootTest.ts:34]: use `withNoneditableRootEditor` if possible

File: modules/tinymce/src/plugins/directionality/test/ts/browser/NoneditableRootTest.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import Plugin from 'tinymce/plugins/directionality/Plugin';
+
+describe('browser.tinymce.plugins.directionality.NoneditableRootTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'directionality',
+    toolbar: 'ltr rtl',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ], true);
+
+  const withNoneditableRootEditor = (editor: Editor, f: (editor: Editor) => void) => {
```
[the remark this answers is not in the document]
Comment (spocke) [reply modules/tinymce/src/plugins/directionality/test/ts/browser/NoneditableRootTest.ts]: Yes, I moved this to mcagar. It was getting out of hand with all duplication. Good point.

**Review by lorenzo-pomili — APPROVED** [review]

**Review by vpyshnenko — APPROVED** [review]

**Pull request merged**